pkg/service/configcache/service.go

```diff
@@ -0,0 +1,217 @@
+// Copyright (C) 2024 ScyllaDB
+
+package configcache
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/scylladb/go-log"
+	"github.com/scylladb/scylla-manager/v3/pkg/scyllaclient"
+	"github.com/scylladb/scylla-manager/v3/pkg/service/cluster"
+	"github.com/scylladb/scylla-manager/v3/pkg/store"
+	"github.com/scylladb/scylla-manager/v3/pkg/util/uuid"
+)
+
+// ConnectionType defines an enum for different types of configuration.
+type ConnectionType int
+
+const (
+	// CQL defines cql connection type.
+	CQL ConnectionType = iota
+	// Alternator defines alternator connection type.
+	Alternator
+
+	updateFrequency = 5 * time.Minute
+)
+
+// ConfigCacher is the interface defining the cache behavior.
+type ConfigCacher interface {
+	Read(clusterID uuid.UUID, host string) (NodeConfig, error)
+	ForceUpdateCluster(ctx context.Context, clusterID uuid.UUID) bool
+	Init(ctx context.Context)
+	Run(ctx context.Context, skipFirst bool)
+}
+
+// Service is responsible for handling all cluster configuration cache related operations.
+// Use svc.Read(clusterID, host) to read the configuration of particular host in given cluster.
+// Use svc.Run() to let the cache update itself periodically with the current configuration.
+//
+// Current implementation doesn't care about changing IP, what means that the configuration for given cluster and host
+// will stay there forever.
+type Service struct {
+	clusterSvc   cluster.Servicer
+	scyllaClient scyllaclient.ProviderFunc
+	secretsStore store.Store
+
+	configs sync.Map
```

Comment:
Because I'm lazy as most of the programmers :) 
`sync.Map` looks to be battle tested solution for concurrent access of map values, it reduces my effort needed to think about correct synchronization (even if it's easy).

The go documentation says:
> The Map type is optimized for two common use cases: (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys. In these two cases, use of a Map may significantly reduce lock contention compared to a Go map paired with a separate Mutex or RWMutex.

In theory we have scenario (2), there are disjoint set of keys.

Reply:
explicit types is something what can be sacrificed, cache interface defines the types explicitly

Reply:
> In theory we have scenario (2), there are disjoint set of keys.

Although I wouldn't say that this map access is on some hot path and this approach still adds some potentially unrecoverable panics to the code, but if you insist on using `sync.Map`, I can live with that:)